umbral/umbral.py

```diff
@@ -120,98 +119,142 @@ def __bytes__(self):
 
-        return eph_e + eph_v + sig
+            return eph_e + eph_v + sig
 
     def verify(self, params: UmbralParameters):
 
-        e = self.point_eph_e
-        v = self.point_eph_v
-        s = self.bn_sig
+        e = self._point_eph_e
+        v = self._point_eph_v
+        s = self._bn_sig
         h = hash_to_bn([e, v], params)
 
         return s * params.g == v + (h * e)
 
     def attach_cfrag(self, cfrag: CapsuleFrag):
         self.cfrags[cfrag.bn_kfrag_id] = cfrag
 
-    def reconstruct(self):
+    def original_components(self):
+        return self._point_eph_e, self._point_eph_v, self._bn_sig
+
+    def _reconstruct(self):
         id_cfrag_pairs = list(self.cfrags.items())
         id_0, cfrag_0 = id_cfrag_pairs[0]
         if len(id_cfrag_pairs) > 1:
-            ids = self.cfrags.keys() 
+            ids = self.cfrags.keys()
             lambda_0 = lambda_coeff(id_0, ids)
             e = lambda_0 * cfrag_0.point_eph_e1
             v = lambda_0 * cfrag_0.point_eph_v1
-            
+
             for id_i, cfrag in id_cfrag_pairs[1:]:
                 lambda_i = lambda_coeff(id_i, ids)
                 e = e + (lambda_i * cfrag.point_eph_e1)
                 v = v + (lambda_i * cfrag.point_eph_v1)
         else:
             e = cfrag_0.point_eph_e1
             v = cfrag_0.point_eph_v1
-        
-        return ReconstructedCapsule(e_prime=e, v_prime=v, x=cfrag_0.point_eph_ni)
 
-    def __bytes__(self):
-        self.to_bytes()
-
-
-class ReconstructedCapsule(object):
-    def __init__(self, e_prime, v_prime, x):
-        self.point_eph_e_prime = e_prime
-        self.point_eph_v_prime = v_prime
-        self.point_eph_ni = x
+        self._point_eph_e_prime = e
+        self._point_eph_v_prime = v
+        self._point_noninteractive = cfrag_0.point_eph_ni
 
-    @staticmethod
-    def from_bytes(data: bytes, curve: ec.EllipticCurve):
+    def _get_contents(self, recp_priv_key: UmbralPrivateKey,
+                      sender_pub_key: UmbralPublicKey,
+                      pre):
         """
-        Instantiate ReconstructedCapsule from serialized data.
+        Reconstructs the Capsule contents from the attached CFrags,
+        opens the Capsule and returns what is inside.
+
+        This will often by a symmetric key.
         """
-        e_prime = Point.from_bytes(data[0:33], curve)
-        v_prime = Point.from_bytes(data[33:66], curve)
-        eph_ni = Point.from_bytes(data[66:99], curve)
+        recp_pub_key = recp_priv_key.get_pub_key(pre.params)
+        self._reconstruct()
 
-        return ReconstructedCapsule(e_prime, v_prime, eph_ni)
+        key = pre.decapsulate_reencrypted(
+            recp_pub_key.point_key, recp_priv_key.bn_key,
+            sender_pub_key.point_key, self
 
-    def to_bytes(self):
+        )
+        return key
+
+    def decrypt(self, recp_priv_key: UmbralPrivateKey,
+                sender_pub_key: UmbralPublicKey,
+                ciphertext: bytes, pre):
```

Comment:
The concept of "sender" is not right here. Anyone that knows the public key could be a potential sender, without being possible to know who he/she was. What we need here is the public key of the original owner (i.e., Alice). I'd propose something like `original_pub_key`, but since we seem to have a convention with names, it could be `alice_pub_key `.

Reply:
Oh, I absolutely agree.  I added "naming conventions" to homecoming discussions.  There are a lot of names all over this project that drive me crazy.  

Rather than try to fix them all (I did fix a few), I was trying to get this PR with the current names and then fix them later.  But maybe it does make sense to fix these now.

Reply:
That's fine. As long as we do it at some moment, it's OK. 
I personally liked the "Alice" name. It's consistent with PRE literature, where re-encryption is usually illustrated from Alice to Bob. Other more technical names (e.g., owner, delegator, etc.) are much uglier, IMHO.